sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -840,3 +840,85 @@ case class GreaterThanOrEqual(left: Expression, right: Expression)
 
   protected override def nullSafeEval(input1: Any, input2: Any): Any = ordering.gteq(input1, input2)
 }
+
+trait BooleanTest extends UnaryExpression with Predicate with ExpectsInputTypes {
+
+  def boolValue: Boolean
+  def whenNull: Boolean
```

Comment:
sorry, but ore more comment... I feel these variable names above are a little ambiguous, so can we have better name? e.g., `boolValueForTest` and `boolValueWhenNull`? (I don't think these are the best though) @dongjoon-hyun might have better idea..

Reply:
+1 for `boolValueWhenNull`. For the other, `boolValueForComparison` might be better instead of `boolValueForTest`?

Reply:
@beliefer ok, let's do `boolValue`->`boolValueForComparison` and `whenNull`-> 
`boolValueWhenNull`

Reply:
@dongjoon-hyun @maropu OK. I will change it.